test/integration/Elsa.Activities.IntegrationTests/Branching/FlowJoinTests.cs

```diff
@@ -0,0 +1,228 @@
+            Activities = { startActivity, flowJoin, afterJoin },
+            Connections = 
+            {
+                new() { Source = new(startActivity, "Done"), Target = new(flowJoin) },
+                new() { Source = new(flowJoin, "Done"), Target = new(afterJoin) }
+            }
+        };
+
+        // Act
+        await RunFlowchartAsync(_services, flowchart);
+
+        // Assert
+        Assert.Contains("Start", _output.Lines);
+        
+        // Document expected behavior for different combinations
+        if (useTokenFlow)
+        {
+            // In token flow mode, both WaitAny and WaitAll should act as no-ops
+            // The flowchart should execute normally: Start -> FlowJoin -> AfterJoin
+            Assert.Contains("AfterJoin", _output.Lines);
+        }
+        else
+        {
+            // In counter flow mode:
+            // - WaitAny should allow continuation after first inbound connection
+            // - WaitAll should wait for all inbound connections (in this case just one)
+            // Since there's only one inbound connection, both should behave similarly
+            Assert.Contains("AfterJoin", _output.Lines);
+        }
+    }
+
+    [Theory]
+    [InlineData(true, FlowJoinMode.WaitAny)]
+    [InlineData(true, FlowJoinMode.WaitAll)]
+    [InlineData(false, FlowJoinMode.WaitAny)]
+    [InlineData(false, FlowJoinMode.WaitAll)]
+    public async Task Should_Handle_Fork_Join_Scenarios(bool useTokenFlow, FlowJoinMode joinMode)
+    {
+        // Test FlowJoin with a Fork-Join pattern
+        // Arrange
+        Flowchart.UseTokenFlow = useTokenFlow;
+
+        var start = new WriteLine("Start");
+        var fork = new FlowFork { Branches = new(["Branch1", "Branch2"]) };
+        var branch1 = new WriteLine("Branch1");
+        var branch2 = new WriteLine("Branch2");
+        var join = new FlowJoin { Mode = new(joinMode) };
+        var afterJoin = new WriteLine("AfterJoin");
+
+        var flowchart = new Flowchart
+        {
+            Start = start,
+            Activities = { start, fork, branch1, branch2, join, afterJoin },
+            Connections =
+            {
+                new() { Source = new(start, "Done"), Target = new(fork) },
+                new() { Source = new(fork, "Branch1"), Target = new(branch1) },
+                new() { Source = new(fork, "Branch2"), Target = new(branch2) },
+                new() { Source = new(branch1, "Done"), Target = new(join) },
+                new() { Source = new(branch2, "Done"), Target = new(join) },
+                new() { Source = new(join, "Done"), Target = new(afterJoin) }
+            }
+        };
+
+        // Act
+        await RunFlowchartAsync(_services, flowchart);
+
+        // Assert
+        Assert.Contains("Start", _output.Lines);
+        
+        if (useTokenFlow)
+        {
+            // In token flow mode, FlowJoin acts as a no-op
+            // Both branches should execute and continue to AfterJoin
```

Comment:
I'm not sure this is entirely correct - even with token mode, the JoinMode property of the FlowJoin activity is significant, because the value is used to derive the `MergeMode`.